Tests/ASCornerLayoutSpecSnapshotTests.mm

```diff
@@ -0,0 +1,215 @@
+
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopRight offsetOption:center childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopRight offsetOption:center childSizeOnly:YES];
+
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomLeft offsetOption:center childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomLeft offsetOption:center childSizeOnly:YES];
+
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomRight offsetOption:center childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomRight offsetOption:center childSizeOnly:YES];
+}
+
+- (void)testCornerSpecForAllLocationsWithInnerOffset {
+  ASCornerLayoutSpecSnapshotTestsOffsetOption inner = ASCornerLayoutSpecSnapshotTestsOffsetOptionInner;
+  
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopLeft offsetOption:inner childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopLeft offsetOption:inner childSizeOnly:YES];
+
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopRight offsetOption:inner childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopRight offsetOption:inner childSizeOnly:YES];
+
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomLeft offsetOption:inner childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomLeft offsetOption:inner childSizeOnly:YES];
+
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomRight offsetOption:inner childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomRight offsetOption:inner childSizeOnly:YES];
+}
+
+- (void)testCornerSpecForAllLocationsWithOuterOffset {
+  ASCornerLayoutSpecSnapshotTestsOffsetOption outer = ASCornerLayoutSpecSnapshotTestsOffsetOptionOuter;
+  
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopLeft offsetOption:outer childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopLeft offsetOption:outer childSizeOnly:YES];
+
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopRight offsetOption:outer childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationTopRight offsetOption:outer childSizeOnly:YES];
+
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomLeft offsetOption:outer childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomLeft offsetOption:outer childSizeOnly:YES];
+
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomRight offsetOption:outer childSizeOnly:NO];
+  [self testCornerSpecWithLocation:ASCornerLayoutLocationBottomRight offsetOption:outer childSizeOnly:YES];
+}
+
+- (void)testCornerSpecWithLocation:(ASCornerLayoutLocation)location
+                      offsetOption:(ASCornerLayoutSpecSnapshotTestsOffsetOption)offsetOption
+                     childSizeOnly:(BOOL)childSizeOnly {
```

Comment:
Let's use the same name that the corner spec uses (`includeCornerForSizeCalculation` or `wrapsCorner` depends on its final name).